### PR TITLE
E-Mail Preview (Quick prototyping)

### PR DESCRIPTION
**Pull request by Xavier-IV**
## Overview

Allow quick mail prototyping with email preview. Coming from my background developing on Rails (`letter_opener` gem), this has been a life/cost saver.

**Take special note on CAUTION section below**

## Usage

Just ensure your ENV is `development`.
```env
NODE_ENV=development
APPLICATION_URL=http://localhost:3000/
```

Then just call

```typescript
import {Mailer} from "./mailer";

const mailOptions = {
    from: 'your-email@gmail.com',
    to: 'recipient@example.com',
    subject: 'New Surau Added',
    html: `<p>Test body</p>`
};

await Mailer.send(mailOptions)
```

Detailed usage can be found in the `spec` I've provided.

https://github.com/farhan-helmy/ratemysurau/assets/14009259/4f714e79-4cfc-4e90-b14d-ebcf556e7646

## Caution ⚠️⚠️⚠️

With this being said, for production, **DO NOT** use `development`.

Otherwise end-user will see the email supposed to be sent, which is not good!

### CODE REVIEW

File: src/server/services/generate-surau-verification.spec.ts [hunk 1]
```diff
@@ -50,24 +48,12 @@ describe('generate-surau-verification', () => {
...
 
-    expect(nodemailer.createTransport).toBeCalledWith({
-      host: process.env.MAIL_HOST,
-      port: process.env.MAIL_PORT,
-      auth: {
-        user: process.env.MAIL_USERNAME,
-        pass: process.env.MAIL_PASSWORD,
-      }
-    })
-    expect(sendMailMock).toHaveBeenCalledTimes(1)
```
Comment (Xavier-IV) [inline src/server/services/generate-surau-verification.spec.ts:70]: Now that the email sending responsibility is offload to `./mailer.ts`, replacing this instead

File: src/server/services/mailer.ts
Comment (Xavier-IV) [inline src/server/services/mailer.ts:1]: This has both instance and static usage.

```typescript
const mailer = new Mailer(mailOptions);
// perform your heavy stuff. And finally,
mailer.send()
```
Or using static.

```typescript
Mailer.send(mailOptions)
```

File: src/server/services/mailer.spec.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      sendMail: sendMailMock,
+    } as never));
+
+    const mailer = new Mailer(mailOptions);
+    await mailer.send()
+
+    expect(process.env.NODE_ENV).toBe('test');
+    expect(sendMailMock).toHaveBeenCalled();
+    expect(sendMailMock).toHaveBeenCalledWith(mailOptions);
+    expect(previewEmail).not.toHaveBeenCalled()
```
Comment (Xavier-IV) [inline src/server/services/mailer.spec.ts:54]: This spec ensure previewEmail **shouldn't** be called on other than `development` node_env.

**Review by Xavier-IV — COMMENTED** [review]
Feel free to take a look or consider if this is beneficial to the project.

**Review by farhan-helmy — APPROVED** [review]
# TIL Very cool stuff, I learn a lot from you man. Thank you so much

**Pull request merged**